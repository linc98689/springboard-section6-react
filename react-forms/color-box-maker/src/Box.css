
.Box{
    margin:20px;
    display:flex;
    align-items: center;
}
.Box-box{
    margin: 7px;
    box-shadow: 2px 2px 3px rgb(56, 53, 53);
}
.Box-delete{
    padding:0.3em 0.5em;
    border: 1px solid black;
    color:rgb(120, 50, 3);
    display: block;
}
