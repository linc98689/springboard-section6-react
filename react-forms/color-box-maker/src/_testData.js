const testBoxes = [
    {color:"c1", width: 1, height:1},
    {color:"c2", width: 2, height:2},
    {color:"c3", width: 3, height:3},
    {color:"c4", width: 4, height:5}
];
const tbox = {id:0, color:"c", width:1, height:1 ,
    deleteBox(id){
        let btn = document.getElementById(id);
        btn.remove();
    }};

export {testBoxes, tbox};