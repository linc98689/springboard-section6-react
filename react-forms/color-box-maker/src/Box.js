import React from "react";
import "./Box.css";

const Box = ({id,color, width, height, deleteBox})=>{
    const handleDelete = (evt) =>{
        deleteBox(evt.target.id);
    };

    //return
    return (
        <div className="Box">
            <div className="Box-box" style={{backgroundColor:color, width, height}}></div>
            <button  id={id} className="Box-delete" onClick={handleDelete}>X</button>
        </div>
    );
}

export default Box;