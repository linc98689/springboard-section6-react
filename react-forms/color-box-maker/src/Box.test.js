import {render,  fireEvent, screen} from "@testing-library/react";
import '@testing-library/jest-dom';
import Box from "./Box";
import {tbox} from "./_testData";

const {id, color, width, height, deleteBox} = tbox;

// smoke test
test("render BoxList", ()=>{
    render(<Box id={id} color={color} 
    width={width} height={height}
    deleteBox={deleteBox}/>);
})

// snap-shot test
test("should not change", ()=>{
    const {asFragment} = render(<Box id={id} color={color} 
        width={width} height={height}
        deleteBox={deleteBox}/>);
    expect(asFragment()).toMatchSnapshot();
})

// test delete button
test("should delete box", ()=>{
    render(<Box id={id} color={color} 
        width={width} height={height}
        deleteBox={deleteBox}/>);
    
    fireEvent.click(screen.getByRole('button'));

    expect(screen.queryByRole("button")).not.toBeInTheDocument();
})
