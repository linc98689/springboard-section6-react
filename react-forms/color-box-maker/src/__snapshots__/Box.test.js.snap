// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should not change 1`] = `
<DocumentFragment>
  <div
    class="Box"
  >
    <div
      class="Box-box"
      style="width: 1px; height: 1px;"
    />
    <button
      class="Box-delete"
      id="0"
    >
      X
    </button>
  </div>
</DocumentFragment>
`;
